### DISCUSSION TITLE
Shared amqp connections between pipelines

### DISCUSSION THREAD

**Discussion by davidPaymefy** [roadrunner-server/roadrunner#1997] [Q&A] [answered]
Hi,

I was just checking jobs workers with amqp driver and I noticed that each pipeline starts a new connection to rabbitmq server. I was wondering if there is a way to share the connection between them pipelines since it's supposed to reduce pressure on the server.

Does it make sense to share the connection? Why or why not?

Thank you.

**Comment by rustatian** [comment, marked as answer]
Hey @davidPaymefy 👋 
No, you can't share connections. Foremost, because one connection failure will lead to complete outage for all pipelines. Connection and channels are independent per-pipeline in RR. So, any problem with pipeline, let's say A, will not involve on pipeline B.
It might be a problem if you create, like, 5000 pipelines, but for a usual use-case of even 100 pipelines this should not be a problem.
So, from my POV, and on my local benchmarks, this won't be an issue 😃
